### PR TITLE
[AIRFLOW-2293] Fix S3FileTransformOperator to work with boto3

### PR DESCRIPTION
**Pull request by sekikn**
Make sure you have checked _all_ steps below.

### JIRA
- [x] My PR addresses the following [Airflow JIRA](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
    - https://issues.apache.org/jira/browse/AIRFLOW-2293
    - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-XXX\], code changes always need a JIRA issue.

### Description
- [x] Here are some details about my PR, including screenshots of any UI changes:

S3FileTransformOperator doen't work for now since it uses a function
which is no longer supported by boto3. This PR replaces it with a
valid function and adds an unit test for this operator.

### Tests
- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

Added tests.operators.test_s3_file_transform_operator.

### Commits
- [x] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

- [x] Passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -146,7 +146,7 @@ def write_version(filename=os.path.join(*['airflow',
 postgres = ['psycopg2-binary>=2.7.4']
 ssh = ['paramiko>=2.1.1', 'pysftp>=0.2.9']
 salesforce = ['simple-salesforce>=0.72']
-s3 = ['boto3>=1.0.0']
+s3 = ['boto3>=1.4.0']
```
Comment (Fokko) [inline setup.py:149]: Are you sure it it is safe to bump this version?

Comment (sekikn) [reply setup.py:149]: Thanks for the review @Fokko! 
In addition to confirming that all tests succeeded, I checked [Boto3's upgrading notes](https://boto3.readthedocs.io/en/latest/guide/upgrading.html) and confirmed that all classes mentioned there are not used (at least directly) in the current codebase. If you have any further concerns, I'll investigate it.

**Comment by bolkedebruin** [comment]
@Fokko any thoughts?

**Review by Fokko — APPROVED** [review]
@bolkedebruin LGTM. @sekikn Thanks, merging.

**Pull request closed without merging**